### PR TITLE
Changing idoverrideuser-* to treat objectClass case insensitively

### PR DESCRIPTION
**Pull request by felipevolpone**
This is import to avoid problems when migrating from olders
versions of IPA and using idoverrideuser-* commands.

https://pagure.io/freeipa/issue/7074

### CODE REVIEW

**Comment by felipevolpone** [comment]
I'm not very familiar with idviews.py related code, so if a change in other parts of freeipa code is needed, please tell me.

**Comment by felipevolpone** [comment]
Hi @flo-renaud, I finally managed to work on this issue again. So, I've updated the PR with another commit, if you think it looks good, I can squash them.

File: ipaserver/plugins/idviews.py [hunk 1]
```diff
@@ -790,8 +791,11 @@ def pre_callback(self, ldap, dn, *keys, **options):
         except errors.NotFound:
             self.obj.handle_not_found(*keys)
 
-        required_object_classes = set(self.obj.object_class)
-        actual_object_classes = set(entry['objectclass'])
+        def __lower(text):
+            return text.lower()
+
+        required_object_classes = set(map(__lower, self.obj.object_class))
+        actual_object_classes = set(map(__lower, entry['objectclass']))
```
Comment (flo-renaud) [inline ipaserver/plugins/idviews.py]: Hi @felipevolpone 
For consistency, you can use the same method as in line 550:

for required_oc in required_object_classes:
    if not self.obj.has_objectclass(entry['objectclass'], required_oc):
        ....

Comment (felipevolpone) [reply ipaserver/plugins/idviews.py]: Thanks for pointing me that :)

**Comment by flo-renaud** [comment]
Hi @felipevolpone 
Thank you for the patch. Works great!

**Comment by stlaz** [comment]
master:

* 286bbb2ab77559f63d10c8c5c4923520cb7d3d0f Changing idoverrideuser-* to treat objectClass case insensitively

**Comment by stlaz** [comment]
@felipevolpone This needs a manual backport for `ipa-4-5`.

**Comment by felipevolpone** [comment]
Backported in PR #1076

**Pull request closed without merging**